qiskit/primitives/statevector_sampler.py

```diff
@@ -0,0 +1,226 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023, 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Statevector Sampler class
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Iterable
+
+import numpy as np
+from numpy.typing import NDArray
+
+from qiskit import ClassicalRegister, QiskitError, QuantumCircuit
+from qiskit.circuit import ControlFlowOp
+from qiskit.quantum_info import Statevector
+
+from .base import BaseSamplerV2
+from .base.validation import _has_measure
+from .containers import (
+    BasePrimitiveOptions,
+    BitArray,
+    PrimitiveResult,
+    PubResult,
+    SamplerPub,
+    SamplerPubLike,
+    make_data_bin,
+)
+from .containers.bit_array import _min_num_bytes
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+@dataclass
+class Options(BasePrimitiveOptions):
+    """Options for execution."""
+
+    shots: int = 100  # TODO: discuss the default number of shots
+    seed: int | np.random.Generator | None = None
+
+
+@dataclass
+class _MeasureInfo:
+    creg_name: str
+    num_bits: int
+    num_bytes: int
+    qreg_indices: list[int]
+
+
+class Sampler(BaseSamplerV2):
+    """
+    Simple implementation of :class:`BaseSamplerV2` with Statevector.
+
+    :Options:
+
+        - **shots** (int) --
+          The number of shots.
+
+        - **seed** (np.random.Generator or int) --
+          Set a fixed seed or generator for the normal distribution.
+    """
+
+    _options_class = Options
+    options: Options
+
+    def __init__(self, *, options: BasePrimitiveOptions | dict[str, Any] | None = None):
```

Comment:
In light of my comments on #11543 , I propose. The main reason I have put the default shots as a class attribute is to discourage people from setting it in the constructor, and instead get them used to using it in the run method. What do you think?

```suggestion
    _DEFAULT_SHOTS = 512
    
    def __init__(self, seed: np.random.Generator | int = 42):
        """"
        Args:
            seed: The random seed or generator used for drawing shots from the final state.
        """"
        self._seed = seed
        
    @property
    def seed(self) -> np.random.Generator | int:
        """The random seed or generator used for drawing shots from the final state."""
        return self._seed
```